elastic/fleet-server
# using actionID+agentID as action results doc id

## What is the problem this PR solves?

Make sure to have max 1 action result per action per agent. This solves the issue of inaccurate counts on Fleet UI.

## How does this PR solve the problem?

Using unique id for action results doc and ignore version confict, this way we can still use a data stream.

## How to test this PR locally

- Start ES, Kibana, Fleet Server, enroll agents.
- Perform action on agents (e.g. reassign, upgrade)
- Verify that `.fleet-actions-results` has documents with id `actionID:agentID`
- With scale testing, make sure duplicate action results per agent don't cause an error (conflicts are ignored)
- After the kibana change is done to use doc count to calculate action results, verify that there are no duplicate action results per agent.

## Design Checklist




## Checklist





## Related issues



Relates https://github.com/elastic/fleet-server/issues/2596
